### DISCUSSION TITLE
skywalking java enhance bootstrap class error:  attempted to change the schema (add/remove fields) error

### DISCUSSION THREAD

**Discussion by leihuazhe** [apache/skywalking#9865] [Q&A] [answered]
skywalking java enhance bootstrap class java.net.InetAddress got class redefinition failed: attempted to change the schema (add/remove fields) error
this is my plugin link: https://github.com/leihuazhe/skylwalking-dns-plugin

Could anyone solved this before? I'm asking for help, thanks.

**Comment by leihuazhe** [comment]
<img width="619" alt="image" src="https://user-images.githubusercontent.com/25452025/198838759-0d0e2ba4-5b73-4f65-998d-185206b08a9b.png">


<img width="717" alt="image" src="https://user-images.githubusercontent.com/25452025/198838747-d23a9065-a2c9-4abb-8e2d-7b25184aa3c4.png">

<img width="1142" alt="image" src="https://user-images.githubusercontent.com/25452025/198838727-77bff5e5-0be8-4414-9b03-8345ce1605d6.png">

**Comment by wu-sheng** [comment, marked as answer]
This would be a very tricky plugin. DNS is widely used in the RPC core, such as gRPC kernel(netty). 
Most likely,this class has been loaded before instrumentation definition.
I can't be sure whether this is your error case though.

**Comment by leihuazhe** [comment]
Okay, I got it.
